pkg/cli/cmd_helpers.go

```diff
@@ -336,3 +338,47 @@ func (factory *executionHooksFactory) postRunEFunc() func(*cobra.Command, []stri
 		return nil
 	}
 }
+
+// updateProjectFileForAlphaGenerate updates the PROJECT file to replace unsupported
+// plugins with a supported version before running `kubebuilder alpha generate`.
+
+func updateProjectFileForAlphaGenerate() error {
+	projectFilePath := "PROJECT"
+
+	content, err := os.ReadFile(projectFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read PROJECT file: %w", err)
+	}
+
+	projectStr := string(content)
+
+	// No need to update if v3 is not found
+	if !strings.Contains(projectStr, "go.kubebuilder.io/v3") {
```

Comment:
Can we ensure that we do the same if it is any of those:
- go.kubebuilder.io/v3
- go.kubebuilder.io/v3-alpha
- go.kubebuilder.io/v2

I mean, in any case we warning and inform the user that we are doing our best to re-scaffold.
Todo so we will need update the PROJECT file by replacing X with go.kubebuilder.io/v4